### PR TITLE
SALTO-4727: StandardValueSets references Support for fetch with changes detection mode

### PR DESCRIPTION
**Pull request by tamtamirr**
Modified the SVS filter to support the upcoming support in CustomObjects fetchWithChangesDetection.

---

As part of testing **CustomObjects** quick fetch, I've noticed that we lose references on **Standard Picklist Fields**, on objects that were updated since the last fetch. I've changed the filter's implementation to support that.

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/standard_value_sets.ts [hunk 1]
```diff
@@ -258,17 +261,13 @@ export const makeFilter = (
...
-              const clone = inst.clone()
-              await resolveTypeShallow(clone, config.elementsSource)
-              return clone
-            })
-            .filter(isInstanceOfType(STANDARD_VALUE_SET))
+        const svsInstances = config.fetchProfile.metadataQuery.isFetchWithChangesDetection()
+          // In fetch with changes detection mode, we need all the StandardValueSets
+          // in order not to lose existing references on fetched CustomObjects.
+          ? await awu(await elementsSource.getAll())
+            .filter(isInstanceOfTypeSync(STANDARD_VALUE_SET))
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/filters/standard_value_sets.ts:268]: 🔥 🔥 🔥

File: packages/salesforce-adapter/src/filters/standard_value_sets.ts [hunk 2]
```diff
@@ -258,17 +261,13 @@ export const makeFilter = (
...
             .toArray()
+          : fetchedSVSInstances
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/filters/standard_value_sets.ts:270]: Is this a performance optimization or is there a reason not to just use the element source in all cases?

**Review by tomsellek — APPROVED** [review]
LGTM!

File: packages/salesforce-adapter/src/filters/standard_value_sets.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/standard_value_sets.ts:270]: Breaking regular fetch is more concerning, that's why I try to keep the separation  between the modes as strict as possible. However, once we have enough confidence in **Fetch With Changes Detection**, we can safely align the logics in many places.

**Pull request merged**